.github/workflows/documentation.yml

```diff
@@ -0,0 +1,54 @@
+# This is a workflow to generate the documentation automatically.
+
+name: Generate Documentation
+
+# Controls when the action will run. Triggers the workflow on push
+# events but only for the master branch
+on:
+  push:
+    branches:
+      - master
+      - feature/AutomaticDocDeployment # this will be removed once it's working
+jobs:
+  # This workflow contains a single job called "build"
+  gen_doc:
+    strategy:
+      matrix:
+        os: [ubuntu-18.04]
+        java: [11]
+    runs-on: ${{ matrix.os }}
+    name: asciidoctor to html
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Load Cache
+        uses: actions/cache@v1.1.2
+        with:
+          path: ~/.gradle/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set up JDK
+        uses: actions/setup-java@v1.3.0
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Run Gradle test
+        run: ./gradlew test
```

Comment:
Not really necessary, but I know why you did it.
Let my objections be noted, but don't let them hinder the merge.

Reply:
Actually we need it to generate the snippets from the Controller Tests.

Reply:
ah fair